packages/class/src/config_utils.ts

```diff
@@ -0,0 +1,139 @@
+import type { Config, JsonSchemaOfConfig } from "./config.js";
+
+/**
+ * Overwrite values in first configuration with second configuration.
+ *
+ * @param first
+ * @param second
+ * @returns Shallow copy of first with values of second.
+ */
+// biome-ignore lint/suspicious/noExplicitAny: recursion is hard to type
+export function mergeConfigurations(first: any, second: any) {
```

Comment:
Would it be better if we keep the config flat but add a 'group' parameter to generate the groups in the web form?